### PR TITLE
Add spike cask

### PR DESCRIPTION
**Pull request by Caesarovich**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
(Error, no cask with such name exists)
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --new-cask <cask>` worked successfully.
(Error, version '3.2.0' differs from '' retrieved by livecheck)
- [x] `brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

Hi, this is my first submission here and I hope I've done all the steps correctly. Please let me know of anything of note.

This casks adds the LEGO Spike software for the new LEGO Spike sets (The ones now replacing Mindstorms and EV3)

### CODE REVIEW

**Comment by bevanjkay** [comment]
Please try to add a [zap stanza](https://docs.brew.sh/Cask-Cookbook#stanza-zap). The linked documentation provides some recommended methods to create it.

**Comment by Caesarovich** [comment]
Is it normal that the brew style --fix commands automatically adds some comments on the top of the file ?

**Comment by bevanjkay** [comment]
Thank you @Caesarovich 
Based on the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference) guidelines the cask token, and filename should just be `spike` not `lego-spike`.

> Is it normal that the brew style --fix commands automatically adds some comments on the top of the file ?
Try running `brew style --fix <token>` instead of `brew style --fix <token>.rb`

File: Casks/spike.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  version "3.2.0"
+  sha256 "9a4aca2d7a8963eb4279ca5c215a828f64bb93c3983497dd18b77ae76b88a677"
+
+  url "https://education.lego.com/_/downloads/SPIKE_APP_3_macOS__#{version}_Global.dmg"
+  name "Lego SPIKE"
+  desc "Develop with Scratch and Python for your LEGO Spike set"
+  homepage "https://education.lego.com/"
+
+  depends_on macos: ">= :catalina"
+  depends_on arch: :x86_64
```
Comment (bevanjkay) [inline Casks/spike.rb:11]: @Caesarovich This likely isn't necessary. I was able to install and run on an M1 Macbook Pro (albeit potentially with Rosetta).

File: Casks/spike.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+cask "spike" do
+  version "3.2.0"
+  sha256 "9a4aca2d7a8963eb4279ca5c215a828f64bb93c3983497dd18b77ae76b88a677"
+
+  url "https://education.lego.com/_/downloads/SPIKE_APP_3_macOS__#{version}_Global.dmg"
```
Comment (bevanjkay) [inline Casks/spike.rb:5]:
```suggestion
  url "https://education.lego.com/_/downloads/SPIKE_APP_#{version.major}_macOS__#{version}_Global.dmg"
```

File: Casks/spike.rb (hunk 1 above)
Comment (Caesarovich) [reply Casks/spike.rb]: Alright thanks, I based this off the system requirements on the page

**Comment by Caesarovich** [comment]
Thanks for all !

**Review by p-linnane — APPROVED** [review]
Thank you @Caesarovich.

**Pull request merged**